clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerHeartbeatRequestManagerTest.java

```diff
@@ -603,29 +606,24 @@ public void testHeartbeatResponseOnErrorHandling(final Errors error, final boole
     }
 
     @Test
-    public void testUnsupportedVersion() {
```

Comment:
Could you please consider using `ParameterizedTest` to eliminate duplicate code? 

```java
    @ParameterizedTest
    @ValueSource(strings = {CONSUMER_PROTOCOL_NOT_SUPPORTED_MSG, REGEX_RESOLUTION_NOT_SUPPORTED_MSG})
    public void testUnsupportedVersion(String errorMessage) {
        mockResponseWithException(new UnsupportedVersionException(errorMessage));
        ArgumentCaptor<ErrorEvent> errorEventArgumentCaptor = ArgumentCaptor.forClass(ErrorEvent.class);
        verify(backgroundEventHandler).add(errorEventArgumentCaptor.capture());
        ErrorEvent errorEvent = errorEventArgumentCaptor.getValue();
        assertInstanceOf(Errors.UNSUPPORTED_VERSION.exception().getClass(), errorEvent.error());
        assertEquals(errorMessage, errorEvent.error().getMessage());
    }
```

Reply:
sure, makes sense, done.